.github/workflows/build-and-test-fedora-34.yml

```diff
@@ -41,6 +41,11 @@ jobs:
         run: ./autogen.sh
       - name: Run `./configure`
         run: ./configure
+      - name: Fix `-pthread` in `GLIB_CFLAGS`
```

Comment:
I'm not proficient enough with `The Autotools` to properly fix this.

Reply:
Here's the error:
```shell
apxs -c   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -pthread  \
	-I./includes  -lmemcached -lrados -lcurl -lglib-2.0  \
		./src/mod_tile.c  \
		./src/sys_utils.c \
		./src/store.c \
		./src/store_file.c \
		./src/store_file_utils.c \
		./src/store_memcached.c \
		./src/store_rados.c \
		./src/store_ro_http_proxy.c \
		./src/store_ro_composite.c \
		./src/store_null.c \
		./src/g_logger.c
apxs:Error: Unknown option: t.
apxs:Error: Unknown option: h.
apxs:Error: Unknown option: r.
apxs:Error: Unknown option: d.
Usage: apxs -g [-S <var>=<val>] -n <modname>
       apxs -q [-v] [-S <var>=<val>] [<query> ...]
       apxs -c [-S <var>=<val>] [-o <dsofile>] [-D <name>[=<value>]]
               [-I <incdir>] [-L <libdir>] [-l <libname>] [-Wc,<flags>]
               [-Wl,<flags>] [-p] <files> ...
       apxs -i [-S <var>=<val>] [-a] [-A] [-n <modname>] <dsofile> ...
       apxs -e [-S <var>=<val>] [-a] [-A] [-n <modname>] <dsofile> ...
```

Reply:
So the `glib-2.0` pkg-config module quite reasonably includes `-pthread` in the compiler flags - it is a valid compiler flag and it does a bit more than just linking the library.

The makefile then passes `$(GLIB_CFLAGS)` when invoking `apxs` but apxs only actually understands a handful of common compiler flags like `-D`, `-I` and `-l` so that is kind of buggy really - other flags need a `-Wc` to tell it they are a compiler bug.

So for apxs you really want `-Wc,-pthread` as the argument.

Reply:
I've opened #249 to fix this, which is a genuine build problem on Fedora...

Reply:
Awesome, thank you for your explanation (and for the fix)! Yeah, I was trying to figure out how to fix this some other way and found it strange that `-pthread` wasn't added on `Ubuntu Latest` and `CentOS 7`.

* Ubuntu Latest:
  ```shell
  apxs -c   -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include \
	  -I./includes  -lmemcached -lrados -lcurl -lglib-2.0 \
		  ./src/mod_tile.c  \
		  ./src/sys_utils.c \
		  ./src/store.c \
		  ./src/store_file.c \
		  ./src/store_file_utils.c \
		  ./src/store_memcached.c \
		  ./src/store_rados.c \
		  ./src/store_ro_http_proxy.c \
		  ./src/store_ro_composite.c \
		  ./src/store_null.c \
		  ./src/g_logger.c
  ```
* CentOS 7:
  ```shell
  apxs -c   -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include   \
	  -I./includes  -lmemcached -lrados -lcurl   -lglib-2.0   \
		  ./src/mod_tile.c  \
		  ./src/sys_utils.c \
		  ./src/store.c \
		  ./src/store_file.c \
		  ./src/store_file_utils.c \
		  ./src/store_memcached.c \
		  ./src/store_rados.c \
		  ./src/store_ro_http_proxy.c \
		  ./src/store_ro_composite.c \
		  ./src/store_null.c \
		  ./src/g_logger.c
  ```